kotsadm/pkg/store/s3pg/reporting_store.go

```diff
@@ -126,3 +150,70 @@ func (s S3PGStore) getK8sVersion() (string, error) {
 	}
 	return k8sVersion.GitVersion, nil
 }
+
+func getKotsadmIDConfigMap() (*corev1.ConfigMap, error) {
+	clientset, err := k8s.Clientset()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get clientset")
+	}
+	namespace := os.Getenv("POD_NAMESPACE")
+	existingConfigmap, err := clientset.CoreV1().ConfigMaps(namespace).Get(context.TODO(), configMapName, metav1.GetOptions{})
+	if err != nil && !kuberneteserrors.IsNotFound(err) {
+		return nil, errors.Wrap(err, "failed to get configmap")
+	} else if kuberneteserrors.IsNotFound(err) {
+		return nil, nil
+	}
+	if existingConfigmap != nil {
+		return existingConfigmap, nil
+	}
+	return nil, nil
+}
+
+// CreateAdminIDConfigMap creates an id for an kotsadm instance and stores in configmap
+func CreateAdminIDConfigMap(clusterID string) (*corev1.ConfigMap, error) {
```

Comment:
this parameter is not always the `clusterID`, i would suggest renaming it to `kotsadmID`